redis/tests/test_sentinel.rs

```diff
@@ -0,0 +1,456 @@
+mod support;
+
+use std::collections::HashMap;
+
+use redis::{
+    sentinel::{Sentinel, SentinelClient, SentinelNodeConnectionInfo},
+    Client, Connection, ConnectionAddr, ConnectionInfo,
+};
+
+use crate::support::*;
+
+fn parse_replication_info(value: &str) -> HashMap<&str, &str> {
+    let info_map: std::collections::HashMap<&str, &str> = value
+        .split("\r\n")
+        .filter(|line| !line.trim_start().starts_with('#'))
+        .filter_map(|line| line.split_once(':'))
+        .collect();
+    info_map
+}
+
+fn assert_is_master_role(replication_info: String) {
+    let info_map = parse_replication_info(&replication_info);
+    assert_eq!(info_map.get("role"), Some(&"master"));
+}
+
+fn assert_replica_role_and_master_addr(replication_info: String, expected_master: &ConnectionInfo) {
+    let info_map = parse_replication_info(&replication_info);
+
+    assert_eq!(info_map.get("role"), Some(&"slave"));
+
+    let (master_host, master_port) = match &expected_master.addr {
+        ConnectionAddr::Tcp(host, port) => (host, port),
+        ConnectionAddr::TcpTls {
+            host,
+            port,
+            insecure: _,
+        } => (host, port),
+        ConnectionAddr::Unix(..) => panic!("Unexpected master connection type"),
+    };
+
+    assert_eq!(info_map.get("master_host"), Some(&master_host.as_str()));
+    assert_eq!(
+        info_map.get("master_port"),
+        Some(&master_port.to_string().as_str())
+    );
+}
+
+fn assert_is_connection_to_master(conn: &mut Connection) {
+    let info: String = redis::cmd("INFO").arg("REPLICATION").query(conn).unwrap();
+    assert_is_master_role(info);
+}
+
+fn assert_connection_is_replica_of_correct_master(conn: &mut Connection, master_client: &Client) {
+    let info: String = redis::cmd("INFO").arg("REPLICATION").query(conn).unwrap();
+    assert_replica_role_and_master_addr(info, master_client.get_connection_info());
+}
+
+/// Get replica clients from the sentinel in a rotating fashion, asserting that they are
+/// indeed replicas of the given master, and returning a list of their addresses.
+fn connect_to_all_replicas(
+    sentinel: &mut Sentinel,
+    master_client: &Client,
+    node_conn_info: &SentinelNodeConnectionInfo,
+    number_of_replicas: u16,
+) -> Vec<ConnectionAddr> {
+    let mut replica_conn_infos = vec![];
+
+    for _ in 0..number_of_replicas {
+        let replica_client = sentinel
+            .replica_rotate_for("master1", Some(node_conn_info))
```

Comment:
Good point. I guess I ignored this part because I was using just "master1" everywhere. But I've added it as a parameter to this function and all similar ones. I've also extracted the string "master1" to a single variable per test (I've thought about a single global one for the whole file, but thought it was best to just replicate it for each test).